src/ProvenanceTransformer.cpp

```diff
@@ -0,0 +1,452 @@
+#include "ProvenanceTransformer.h"
+#include "AstVisitor.h"
+
+namespace souffle {
+
+/**
+ * Helper functions
+ */
+const std::string& identifierToString(const AstRelationIdentifier& name) {
```

Comment:
Don't return a reference to a temporary object, which the string from the AstRelationIdentifier always will be. Just return a new std::string